### DISCUSSION TITLE
Set multi-env file for specific buildType

### DISCUSSION THREAD

**Discussion by yepMad** [goatandsheep/react-native-dotenv#412] [Q&A]
Hello,

Is it possible to define which envfile will be read by a specific buildType? Why, for example: on Android I have the buildType "releaseStaging", I imagine that the file read by it is ".env.production"

**Comment by goatandsheep** [comment]
Yes :) https://github.com/goatandsheep/react-native-dotenv/wiki/Multi-env-troubleshooting#android make sure you set the right mode

**Reply by yepMad** [reply]
Thanks, @goatandsheep !! Sorry to bother you, but I'm not sure I understand correctly. How can I make my buildType “releaseStaging” read “.env.staging”? I don't know if I can understand how to do this exactly.

**Reply by goatandsheep** [reply]
Good question. I haven't really tested mode besides development and production on Android. I'll try testing it again but if you can't figure it out before I test it, try building using APP_ENV=staging

But in general if you're choosing .env.production you need to set your config to use mode=production

Basically if the value of NODE_ENV=staging it should read .env.staging if it's configured correctly

**Reply by yepMad** [reply]
I think it might work, I haven't been able to test it yet. But my fear is that this will affect something, I mean, if there is any module that tests _"if (process.env.NODE_ENV !== "production")"_ we will be in trouble.

Or do you think this shouldn't be a concern?

**Reply by goatandsheep** [reply]
It should be safe. Otherwise you can add something like a console.log there and you'll see.
